### PR TITLE
Add a callback for connection close

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: amqp.go [hunk 1]
```diff
@@ -81,6 +89,12 @@ func NewExchange(url string, options *ExchangeOptions) (*Exchange, error) {
 		return nil, err
 	}
 
+	if options.OnDisconnect == nil {
+		options.OnDisconnect = func() bool {
+			panic("RabbitMQ Disconnected")
+		}
```
Comment (bmarini) [inline amqp.go]: This is basically the same as the default queue option one above, might as well do this:

``` go
var DefaultOnDisconnect = func() bool {
    panic("RabbitMQ disconnected")
    return false
}
```

and use that in both places.

Comment (ghost) [inline amqp.go]: Whoops yeah forgot to remove this.

**Comment by bmarini** [comment]
This looks good. I wonder if there are any other code paths though where rabbitmq is disconnected. I'd expect this func to get called for all of them.

**Comment by ghost** [comment]
There's NotifyClose here https://github.com/streadway/amqp/blob/c8da4cb5f60d433d666e5c76af18919c51048451/channel.go#L411 but I think we should still just check the Acknowledger. I think it's guaranteed to run, and if we use NotifyClose then we can't call Handler.Stop() (due to channels being closed by the time NotifyClose is reached), which is needed to clean up some goroutines.

**Pull request merged**